### ISSUE TITLE
pkg.installed no longer accepts a version of "latest"

### ISSUE THREAD

**Issue by paul-mulvihill** [saltstack/salt#42821]
### Description of Issue/Question
It looks like using the "version: latest" feature on pkg.installed no longer works properly.
On a test run it identifies the right package (latest from repo) although when actually applying it looks for package-latest which isn't found.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
php:
  pkg.installed:
    - name: php
    - version: {{ salt.pillar.get('versions:RedHat:seven:php', 'latest') }}

### Steps to Reproduce Issue
Try to run the package install without a matching pillar information.
Running it on test=True will show the version it will try to install. Although when it actually tries to apply the state and install the latest it reports back saying the package php-latest is not found.

[root@sltdev01 ~]# salt 'testserver' state.apply php test=True
testserver:
----------
          ID: php
    Function: pkg.installed
      Result: None
     Comment: The following packages would be installed/updated: php=5.4.16-42.el7
     Started: 15:12:10.806689
    Duration: 3618.468 ms
     Changes:

Summary for testserver
------------
Succeeded: 1 (unchanged=1)
Failed:    0
------------
Total states run:     1
Total run time:   3.618 s

[root@sltdev01 ~]# salt 'testserver' state.apply php
testserver:
----------
          ID: php
    Function: pkg.installed
      Result: False
     Comment: Error occurred installing package(s). Additional info follows:

              errors:
                  - Running scope as unit run-3184.scope.
                    Loaded plugins: fastestmirror, priorities
                    Loading mirror speeds from cached hostfile
                     * base: mirror.sov.uk.goscomb.net
                     * extras: centos.mirroring.pulsant.co.uk
                     * updates: centos.mirroring.pulsant.co.uk
                    No package php-latest available.
                    Error: Nothing to do
     Started: 15:12:23.911264
    Duration: 12255.069 ms
     Changes:

### Versions Report
Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: 3.4.3
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.3.1611 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-514.26.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core

**Comment by gtmanfred** [comment]
This is a duplicate of https://github.com/saltstack/salt/issues/42194

This will be fixed in a future version.  I don't think it made it into 2017.7.1, but will definitely be in 2017.7.2

Thanks,
Daniel

**Comment by paul-mulvihill** [comment]
Do you have some dates for when these two releases will be uploaded to the public repo?
I've not seen any sign on 2017.7.1 on your repo yet and 2017.7.02 has the fixes I'm waiting on.
Is there a calender page I can follow to find out?

**Comment by gtmanfred** [comment]
We are working on 2017.7.1, and 2017.7.2 should be sometime next month.

You can also drop in the pkg state module using dynamic modules to get it patched now.

https://docs.saltstack.com/en/latest/ref/file_server/dynamic-modules.html

https://github.com/saltstack/salt/blob/2017.7/salt/states/pkg.py

Thanks,
Daniel
